core/slice.go

```diff
@@ -840,7 +840,7 @@ func (sl *Slice) ConstructLocalMinedBlock(wo *types.WorkObject) (*types.WorkObje
 			sl.hc.chainHeadFeed.Send(ChainHeadEvent{sl.hc.CurrentHeader()})
 			return nil, ErrBodyNotFound
 		}
-		if len(pendingBlockBody.OutboundEtxs()) == 0 {
+		if wo.NumberU64(common.ZONE_CTX) > uint64(params.WorkSharesInclusionDepth) && len(pendingBlockBody.OutboundEtxs()) == 0 {
```

Comment:
Can you explain this addition? Or add comments

Reply:
there will not be any etx for the first three blocks. because we look back 3 blocks